module/mailnews/mailread.c

```diff
@@ -353,3 +404,174 @@ void UserMailReply(int msg_num, Bool reply_all)
+
+   // Extract the msg_num values from the list view items
+   ListView_GetItemText(hListView, lParam1, COL_ORDER, szBuf1, sizeof(szBuf1));
+   ListView_GetItemText(hListView, lParam2, COL_ORDER, szBuf2, sizeof(szBuf2));
+   int msg_num1 = atoi(szBuf1);
+   int msg_num2 = atoi(szBuf2);
+
+   ListView_GetItemText(hListView, lParam1, nColumn, szBuf1, sizeof(szBuf1));
+   ListView_GetItemText(hListView, lParam2, nColumn, szBuf2, sizeof(szBuf2));
+
+   switch (nColumn)
+   {
+   case COL_ORDER:
+      return bSortAscending ? atoi(szBuf1) - atoi(szBuf2) : atoi(szBuf2) - atoi(szBuf1);
+   case COL_FROM:
+      return bSortAscending ? stricmp(szBuf1, szBuf2) : stricmp(szBuf2, szBuf1);
+   case COL_SUBJECT:
+      return bSortAscending ? stricmp(szBuf1, szBuf2) : stricmp(szBuf2, szBuf1);
+   case COL_TIME:
+      time1 = mail_date_map[msg_num1];
+      time2 = mail_date_map[msg_num2];
+
+      return bSortAscending ? (time1 - time2) : (time2 - time1);
+   default:
+      debug(("Unhandled column (column #%d) in CompareListItems\n", nColumn));
+      return 0;
+   }
+}
+
+void ListView_SetHeaderSortImage(HWND hListView, int sortedColumn, BOOL sortAscending)
+{
+   // Get the handle to the header control associated with the list view
+   HWND hHeader = ListView_GetHeader(hListView);
+   if (!hHeader)
+      return;
+
+   int columnCount = Header_GetItemCount(hHeader);
+
+   for (int i = 0; i < columnCount; i++)
+   {
+      HDITEM hdi = {0};
+      hdi.mask = HDI_FORMAT | HDI_BITMAP;
+
+      // Get the current format of the header item
+      Header_GetItem(hHeader, i, &hdi);
+
+      if (i == sortedColumn)
+      {
+         hdi.fmt |= (HDF_BITMAP | HDF_BITMAP_ON_RIGHT);
+         hdi.hbm = sortAscending ? hbmUpArrow : hbmDownArrow;
+      }
+      else
+      {
+         hdi.fmt &= ~(HDF_BITMAP | HDF_BITMAP_ON_RIGHT);
+         hdi.hbm = NULL;
+      }
+
+      // Set the header item at the current index with the updated details
+      Header_SetItem(hHeader, i, &hdi);
+   }
+}
+
+/*
+ * Resets the news list to its default sort: by order, descending
+ */
+void ResetListSort(HWND hListView)
+{
+   // 1-based column
+   ListView_SortItemsEx(hListView, CompareListItems, -(COL_ORDER + 1));
+   // 0-based column
+   ListView_SetHeaderSortImage(hListView, COL_ORDER, FALSE);
+}
+
+int StringToTimestamp(const char *dateStr)
+{
+   struct tm tm = {0}; // Initialize tm to zero
+   char monthStr[4];   // To hold the abbreviated month name
+
+   // Parse the date-time string
+   if (sscanf(dateStr, "%*3s %3s %d, %d %d:%d",
+              monthStr, &tm.tm_mday, &tm.tm_year, &tm.tm_hour, &tm.tm_min) != 5)
+   {
+      // Handle parse error
+      debug(("Failed to parse date string: %s\n", dateStr));
+      return 0;
+   }
+
+   // Adjust the struct tm fields
+   tm.tm_year -= 1900; // Year since 1900
+
+   // Convert month abbreviation to month number
+   const char *months[] = {"Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"};
```

Comment:
This will break in German or any other language---you definitely don't want to be doing this yourself.  Look for a system routine that does this.  Maybe https://stackoverflow.com/questions/321849/strptime-equivalent-on-windows

Reply:
I've checked in what I think might be a viable solution, but have ventured into unfamiliar territory yet again. The solution I dropped in is the most succinct solution I could come up with, this late in the evening at least. I may return to solution in this SO post again tomorrow and see about integrating locale into the mix there.